packaging/examples/cruise-control/kafka-cruise-control-auto-rebalancing.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    version: 3.8.0
+    replicas: 3
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
+      inter.broker.protocol.version: "3.8"
+    storage:
+      type: ephemeral
+  zookeeper:
+    replicas: 3
+    storage:
+      type: ephemeral
+  entityOperator:
+    topicOperator: {}
+    userOperator: {}
+  cruiseControl:
+    autoRebalance:
+      - mode: add-brokers
+        template:
+          name: my-add-brokers-rebalancing-template
+      - mode: remove-brokers
+        template:
+          name: my-remove-brokers-rebalancing-template
+---
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaRebalance
+metadata:
+  name: my-add-brokers-rebalancing-template
+  labels:
+    strimzi.io/cluster: my-cluster
```

Comment:
What is the role of the cluster label? I would expect the template to be used without the label. But right now, using it without the label, it produces following error condition:

```
    - lastTransitionTime: "2024-09-20T15:56:44.736828689Z"
      message: 'Resource lacks label ''strimzi.io/cluster'': No cluster related to
        a possible rebalance.'
      reason: InvalidResourceException
      status: "True"
      type: NotReady
```

And related warning in logs.

Reply:
Good catch!! Of course the label is needed for actual `KafkaRebalance` but this check should be skipped if it's a template.

Reply:
Actually the `KafkaRebalanceAssemblyOperator.reconcileKafkaRebalance` should skip a template at the beginning of the reconciliation not checking for the label. I will take a look.

Reply:
@scholzj I tried it but it works on my side. I pushed the YAML example without the label and the templates are "skipped" as expected so the status is not updated, no error/exception reported there. I also fixed the corresponding test `KafkaRebalanceAssemblyOperatorTest.testRebalanceTemplate` producing the KR without the cluster label and it is green as expected. Maybe any mistake from your side, can you try again to reproduce it?

Reply:
You are right -> that was my mistake. But I'm still getting this every reocnciliation:
```
2024-09-23 20:31:37 INFO  AbstractOperator:266 - Reconciliation #146(timer) KafkaRebalance(myproject/autoscaling-template): KafkaRebalance autoscaling-template will be checked for creation or modification
2024-09-23 20:31:37 INFO  AbstractOperator:546 - Reconciliation #146(timer) KafkaRebalance(myproject/autoscaling-template): reconciled
```

Can we get rid of that as well?

Reply:
But these two lines are coming from the `AbstractOperator` AFAICS as the begin/end of the reconciliation of a general custom resource (in this case a `KafkaRebalance` one by the `KafkaRebalanceAssemblyOperator`, inheriting from `AbstractOperator`). I don't think I can get rid of them easily.

Reply:
Hmm, ok ... :-/